hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieInstant.java

```diff
@@ -70,32 +100,38 @@ public enum State {
     NIL
   }
 
-  private State state = State.COMPLETED;
-  private String action;
-  private String timestamp;
+  private final State state;
+  private final String action;
+  private final String timestamp;
+  private final String stateTransitionTime;
 
   /**
    * Load the instant from the meta FileStatus.
    */
   public HoodieInstant(FileStatus fileStatus) {
     // First read the instant timestamp. [==>20170101193025<==].commit
     String fileName = fileStatus.getPath().getName();
-    String fileExtension = getTimelineFileExtension(fileName);
-    timestamp = fileName.replace(fileExtension, "");
-
-    // Next read the action for this marker
-    action = fileExtension.replaceFirst(".", "");
-    if (action.equals("inflight")) {
-      // This is to support backwards compatibility on how in-flight commit files were written
-      // General rule is inflight extension is .<action>.inflight, but for commit it is .inflight
-      action = "commit";
-      state = State.INFLIGHT;
-    } else if (action.contains(HoodieTimeline.INFLIGHT_EXTENSION)) {
-      state = State.INFLIGHT;
-      action = action.replace(HoodieTimeline.INFLIGHT_EXTENSION, "");
-    } else if (action.contains(HoodieTimeline.REQUESTED_EXTENSION)) {
-      state = State.REQUESTED;
-      action = action.replace(HoodieTimeline.REQUESTED_EXTENSION, "");
+    Matcher matcher = NAME_FORMAT.matcher(fileName);
+    if (matcher.find()) {
+      timestamp = matcher.group(1);
+      if (matcher.group(2).equals(HoodieTimeline.INFLIGHT_EXTENSION)) {
+        // This is to support backwards compatibility on how in-flight commit files were written
+        // General rule is inflight extension is .<action>.inflight, but for commit it is .inflight
+        action = "commit";
+        state = State.INFLIGHT;
+      } else {
+        action = matcher.group(2).replaceFirst(".", "");
+        if (matcher.groupCount() == 3 && matcher.group(3) != null) {
+          state = State.valueOf(matcher.group(3).replaceFirst(".", "").toUpperCase());
+        } else {
+          // Like 20230104152218702.commit
+          state = State.COMPLETED;
+        }
+      }
+      stateTransitionTime =
```

Comment:
I am worried that the modification time will change when you say copy over the table to a new location .. Then all the incremental queries need to be restarted?

Reply:
It is acceptable, because the table path should have changed, it is a fresh new table, the downstream pipelines should have some awareness.

Reply:
but the checkpoints will be lost and the downstream pipelines will have to be recomputed all over again. Have you run into this situation @boneanxs ?

Reply:
Thank you, @vinothchandar , for bringing up this important question. We apologize for not considering this aspect earlier. Since the timeline of the new migrated table could be in a random order if still based on `modificationTime`, it can be challenging for downstream pipelines to recover from a new specified time if it's from the old location but not read yet.

So to migrate a table (let's call it A), I think we can follow these steps:

1. Stop ingesting new data into table A.
2. Wait for all downstream jobs to finish their incremental jobs from table A.
3. Migrate table A to a new location.
4. Delete checkpoint information for all downstream jobs and specify `hoodie.datasource.streaming.startOffset` to the time when the migration is completed. Then restart the jobs.